### PR TITLE
common: improve unit test coverage of 'common' package

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <chris@lonng.org>

### What problem does this PR solve? 

Improve unit test coverage of 'common' package

### CODE REVIEW

**Comment by lonng** [comment]
@kennytm @GregoryIan PTAL

File: lightning/common/util_test.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	c.Assert(err, IsNil)
+	c.Assert(request, DeepEquals, response)
+
+	// Mock `StatusNoContent` response
+	handle = func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(http.StatusNoContent)
+	}
+	err = common.GetJSON(http.DefaultClient, testServer.URL, &response)
+	c.Assert(err, NotNil)
+	c.Assert(strings.Contains(err.Error(), "http status code != 200"), IsTrue)
```
Comment (kennytm) [inline lightning/common/util_test.go]: Combine these two lines (79–80) into:

```suggestion
	c.Assert(err, ErrorMatches ".*http status code != 200.*")
```

Comment (kennytm) [reply lightning/common/util_test.go]: @lonng I mean `ErrorMatches` already implies `NotNil`. https://github.com/pingcap/check/blob/67f458068fc864dabf17e38d4d337f28430d13ed/checkers.go#L260-L262

**Comment by kennytm** [comment]
PTAL @july2993

**Pull request merged**